pilot/pkg/networking/core/listener_waypoint.go

```diff
@@ -640,6 +633,101 @@ func (lb *ListenerBuilder) buildWaypointInboundHTTPFilters(svc *model.Service, c
 	return filters
 }
 
+// buildInboundNetworkFilters generates a TCP proxy network filter on the inbound path
+func (lb *ListenerBuilder) buildWaypointNetworkFilters(svc *model.Service, fcc inboundChainConfig) []*listener.Filter {
```

Comment:
nit: feels kinda bad that we're fragmenting the waypoint implementation even more now. Probably not avoidable in the short term, but something to think about

Reply:
In another PR I am de-fragmenting the HTTP path so hopefully I am net-even or positive for the week :slightly_smiling_face: .

But I generally agree